core/state.py

```diff
@@ -2023,6 +2030,12 @@ def SetNamed(self, lval, val, which_scopes, flags=0):
         This can be used to FLIP flags, while NOT changing the variable.  Or it
         can also be used to initialize?
         """
+
+        # Hack: let SearchPath know it needs to reset the path
```

Comment:
I think the assumption here is that this is the ONLY way to change PATH

We also have an issue with `LANG=C` -- in bash, that doesn't just mutate a variable, it also makes a function call like `setlocale()` !!